### PR TITLE
Add iscsi-direct pool to valid pool type

### PR DESCRIPTION
**Pull request by meinaLi**
Add iscsi-direct pool to valid pool type
Signed-off-by: Meina Li <meili@redhat.com>

### CODE REVIEW

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -1027,22 +1027,23 @@ def pre_pool(self, pool_name, pool_type, pool_target, emulated_image,
                 source_host = self.params.get("source_host", "localhost")
                 extra = "--source-host %s --source-path %s" % (source_host,
                                                                nfs_path)
-        elif pool_type == "iscsi":
+        elif pool_type == "iscsi" or pool_type == "iscsi-direct":
```
Comment (Yingshun) [inline virttest/utils_test/libvirt.py]: suggest update 'if' condition to 'if pool_type in ["iscsi", "iscsi-direct"]'.

File: virttest/utils_test/libvirt.py [hunk 2]
```diff
@@ -1027,22 +1027,23 @@ def pre_pool(self, pool_name, pool_type, pool_target, emulated_image,
...
             iscsi_chap_password = kwargs.get('iscsi_chap_password', None)
             iscsi_secret_usage = kwargs.get('iscsi_secret_usage', None)
+            iscsi_initiator = kwargs.get('iscsi_initiator', None)
             if ip_protocal == "ipv6":
                 ip_addr = "::1"
             else:
                 ip_addr = "127.0.0.1"
             if iscsi_chap_user and iscsi_chap_password and iscsi_secret_usage:
-                logging.debug("setup iscsi pool with chap authentication")
+                logging.debug("setup iscsi/iscsi-direct pool with chap authentication")
```
Comment (Yingshun) [inline virttest/utils_test/libvirt.py]: logging.debug("setup %s pool with chap authentication", pool_type)

File: virttest/utils_test/libvirt.py [hunk 3]
```diff
@@ -1027,22 +1027,23 @@ def pre_pool(self, pool_name, pool_type, pool_target, emulated_image,
...
                 extra = (" --auth-type chap --auth-username %s "
                          "--secret-usage %s" %
                          (iscsi_chap_user, iscsi_secret_usage))
             else:
-                logging.debug("setup iscsi pool without authentication")
+                logging.debug("setup iscsi/iscsi-direct pool without authentication")
```
Comment (Yingshun) [inline virttest/utils_test/libvirt.py]: as above.

File: virttest/utils_test/libvirt.py [hunk 4]
```diff
@@ -1056,8 +1057,14 @@ def pre_pool(self, pool_name, pool_type, pool_target, emulated_image,
...
             iscsi.iscsi_logout(iscsi_target)
-            extra += " --source-host %s  --source-dev %s" % (ip_addr,
-                                                             iscsi_target)
+            if pool_type == "iscsi-direct":
+                extra += " --source-host %s --source-dev %s " \
+                         "--source-iqn %s" % (ip_addr, iscsi_target,
+                                              iscsi_initiator)
+            else:
+                extra += " --source-host %s  --source-dev %s" % (ip_addr,
+                                                                 iscsi_target)
```
Comment (Yingshun) [inline virttest/utils_test/libvirt.py]: I think only add --source-iqn for iscsi-direct pool is fine here.
            if pool_type == "iscsi-direct":
                extra += "--source-iqn %s" %  iscsi_initiator

File: virttest/utils_test/libvirt.py (hunk 1 above)
Comment (meinaLi) [reply virttest/utils_test/libvirt.py]: Updated

File: virttest/utils_test/libvirt.py (hunk 2 above)
Comment (meinaLi) [reply virttest/utils_test/libvirt.py]: Updated

File: virttest/utils_test/libvirt.py (hunk 4 above)
Comment (meinaLi) [reply virttest/utils_test/libvirt.py]: Updated, thaks for your useful suggestions.

**Comment by meinaLi** [comment]
> This pull request **introduces 1 alert** when merging [8a87fe7](https://github.com/avocado-framework/avocado-vt/commit/8a87fe7e376212b3f69efd78e955bfbbee5f1ab4) into [91367aa](https://github.com/avocado-framework/avocado-vt/commit/91367aad95b3f848d679fe46c8a262c481d39c69) - [view on LGTM.com](https://lgtm.com/projects/g/avocado-framework/avocado-vt/rev/pr-63349a3b225e79cda481207fdd12890869d89bfe)
> 
> **new alerts:**
> 
> * 1 for Wrong number of arguments in a call
No need to resolve this alert

**Comment by smitterl** [comment]
`--source-iqn` is not available. Related BZ https://bugzilla.redhat.com/show_bug.cgi?id=1658082 (DEFERRED). That is, usage of this option is wrong. Also related https://github.com/autotest/tp-libvirt/pull/2337

**Pull request merged**